tests/ProjectsMetadataTest.php

```diff
@@ -0,0 +1,188 @@
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => $this->normalUser['email']]);
+
+        $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+    }
+
+    public function testOrgAdminCanAddMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+
+        $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+    }
+
+
+    public function allowedAddMetadataRoles(): array
+    {
+        return [
+            [
+                ProjectRole::ADMIN,
+            ],
+            [
+                ProjectRole::SHARE,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider allowedAddMetadataRoles
+     */
+    public function testProjectMemeberCanAddMetadata(string $role)
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        $this->client->addUserToProject(
+            $projectId,
+            [
+                'email' => $this->normalUser['email'],
+                'role' => $role,
+            ]
+        );
+
+        $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+    }
+
+    public function notAllowedAddMetadataRoles(): array
+    {
+        return [
+            [
+                ProjectRole::GUEST,
+            ],
+            [
+                ProjectRole::READ_ONLY,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider notAllowedAddMetadataRoles
+     */
+    public function testProjectMemeberCannotAddMetadata(string $role)
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
+        $projectId = $this->createProjectWithSuperAdminMember($this->organization['id']);
+
+        $this->client->addUserToProject(
+            $projectId,
+            [
+                'email' => $this->normalUser['email'],
+                'role' => $role,
+            ]
+        );
+
+        try {
+            $this->normalUserClient->setProjectMetadata($projectId, self::PROVIDER_USER, self::TEST_METADATA);
+            $this->fail('Should fail.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+    }
+
+    public function testSuperAdminWithoutMfaCannotAddMetadata()
```

Comment:
Tohle mi přijde hrozně neočekávané - sice může z venku něco měnit když mají zakázaný autojoin, ale nemůžu to dělat pokud mají requireMfa ....

Reply:
No takto to funguje napr u tych limitov, s @ErikZigo som si okolo toho volal, ako to je potom asi na dalsiu diskusiu co to ma takto fungovat alebo ne. Ale imho minimalne teraz hej pretoze to by sa pak muselo upravit aj prelimity a pripadne dalsie